### PR TITLE
Update to ESLint 4.0

### PR DESCRIPTION
**Pull request by lo1tuma**

### CODE REVIEW

File: baseConfig.json [hunk 1]
```diff
@@ -181,6 +181,8 @@
         "no-magic-numbers": "off",
         "no-mixed-operators": "off",
         "array-bracket-spacing": [ "error", "always" ],
+        "array-bracket-newline": "off",
+        "array-element-newline": [ "error", { "multiline": true } ],
```
Comment (selaux) [inline baseConfig.json]: Does this mean that arrays containing non-multiline elements are not allowed to have newlines? This is not clear to me from the eslint rule examples...

File: baseConfig.json [hunk 2]
```diff
@@ -291,6 +290,15 @@
...
         "sort-keys": "off",
         "symbol-description": "off",
-        "template-tag-spacing": "off"
+        "template-tag-spacing": "off",
+        "for-direction": "off",
+        "no-buffer-constructor": "off",
+        "padding-line-between-statements": [
+            "error",
+            { "blankLine": "always", "prev": "directive", "next": "*" },
+            { "blankLine": "any", "prev": "directive", "next": "directive" }
```
Comment (selaux) [inline baseConfig.json]: Is there ever the case where a directive follows a directive?

File: baseConfig.json [hunk 3]
```diff
@@ -291,6 +290,15 @@
         "require-await": "off",
         "sort-keys": "off",
         "symbol-description": "off",
-        "template-tag-spacing": "off"
+        "template-tag-spacing": "off",
+        "for-direction": "off",
+        "no-buffer-constructor": "off",
```
Comment (selaux) [inline baseConfig.json]: Does it maybe make sense to enable this to catch common errors?

File: baseConfig.json (hunk 1 above)
Comment (lo1tuma) [inline baseConfig.json]: If an array literal spans over multiple lines than this rule requires each **element** of the array to be on a separate line.

Examples:

```js
// good
foo = [ 1, 2, 3, 4 ];
// good
foo = [
  1,
  2,
  3,
  4
];
// bad
foo = [
  1, 2, 3, 4
];
```

Comment (selaux) [inline baseConfig.json]: 👍

File: baseConfig.json (hunk 2 above)
Comment (lo1tuma) [inline baseConfig.json]: I don’t think so. However this is the recommended equivalent of the deprecated rule `lines-around-directive` with settings `[ "error", "always" ]` which we had enabled before.

File: baseConfig.json (hunk 3 above)
Comment (lo1tuma) [inline baseConfig.json]: I think it would make sense only for nodejs environments. Currently we don’t provide a nodejs specific environment.

**Pull request merged**